cmd/abapEnvironmentCheckoutBranch.go

```diff
@@ -150,3 +146,85 @@ func triggerCheckout(repositoryName string, branchName string, checkoutConnectio
 	uriConnectionDetails.URL = body.Metadata.URI + expandLog
 	return uriConnectionDetails, nil
 }
+
+func checkCheckoutBranchRepositoryConfiguration(options abapEnvironmentCheckoutBranchOptions) error {
+	if options.Repositories == "" && options.RepositoryName == "" && options.BranchName == "" {
+		return fmt.Errorf("Checking configuration failed: %w", errors.New("You have not specified any repository configuration to be pulled into the ABAP Environment System. Please make sure that you specified the repositories with their branches that should be pulled either in a dedicated file or via in-line configuration. For more information please read the User documentation"))
+	}
+	if options.Repositories != "" && options.RepositoryName != "" && options.BranchName != "" {
+		log.Entry().Info("It seems like you have specified both the repositories with their branches to be pulled as an in-line configuration as well as in the dedicated repositories configuration file.")
+		log.Entry().Info("Please note that in this case the dedicated repositories configuration file will be handled with priority.")
+	}
+	if options.Repositories != "" && ((options.RepositoryName == "") != (options.BranchName == "")) {
+		log.Entry().Info("It seems like you have specified a dedicated repository configuration file but also an in-line configuration for the repository or branch to be pulled.")
+		log.Entry().Info("Please note that in this case the dedicated repositories configuration file will be handled only.")
+	}
+	return nil
+}
+
+func checkoutBranchesFromConfigFile(repositoriesFile string, checkoutConnectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, pollIntervall time.Duration) (err error) {
+	if fileExists(repositoriesFile) {
+		fileContent, err := ioutil.ReadFile(repositoriesFile)
+		if err != nil {
+			return fmt.Errorf("Failed to read repository configuration file: %w", err)
+		}
+		var repositoriesFileConfig []abaputils.Repository
+		var result []byte
+		result, err = yaml.YAMLToJSON(fileContent)
+		if err == nil {
+			err = json.Unmarshal(result, &repositoriesFileConfig)
+		}
+		if err != nil {
+			return fmt.Errorf("Failed to parse repository configuration file: %w", err)
+		}
+		if len(repositoriesFileConfig) == 0 {
+			return fmt.Errorf("Failed to parse repository configuration file: %w", errors.New("Empty or wrong configuration file. Please make sure that you have correctly specified the branches in the repositories to be checked out"))
+		}
+		for _, repositoryFileConfig := range repositoriesFileConfig {
+			err = handleCheckout(repositoryFileConfig, checkoutConnectionDetails, client, pollIntervall)
+			if err != nil {
+				return fmt.Errorf("Failed to checkout branch: %w", err)
+			}
+		}
+	} else {
+		return fmt.Errorf("Failed to read repository configuration file: %w", errors.New(repositoriesFile+" is not a file or doesn't exist"))
+	}
+	return err
```

Comment:
You have early returns for all errors (which I like), but you can make it clear that everything went fine when reaching here. (I think you don't need the named return value after reversing the if.)

```suggestion
	return nil
```

Reply:
See above.